metafacture-io/src/main/java/org/metafacture/io/SruOpener.java

```diff
@@ -0,0 +1,241 @@
+        this.query = URLEncoder.encode(query, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Sets total number of records to be retrieved. <strong>Default value: indefinite (as in "all")
+     * </strong>.
+     *
+     * @param totalrecords total number of records to be retrieved
+     */
+    public void setTotal(final int totalrecords) {
+        totalRecords = totalrecords;
+    }
+
+    /**
+     * Sets the maximum of records returned in one lookup. <strong>Default value: {@value MAXIMUM_RECORDS}</strong>.
+     * The lookup is repeated as long as {@link #maximumRecords} is less than {@link #totalRecords}.
+     *
+     * @param maximumrecords maximum of records returned in one lookup
+     */
+    public void setMaximumRecords(final int maximumrecords) {
+        maximumRecords = maximumrecords;
+    }
+
+    /**
+     * Sets where to start when retrieving records. <strong>Default value: {@value START_RECORD}</strong>.
+     *
+     * @param startrecord where to start when retrieving records
+     */
+    public void setStartRecord(final int startrecord) {
+        startRecord = startrecord;
+    }
+
+    /**
+     * Sets the format of the retrieved record data. <strong>Default value: {@value RECORD_SCHEMA}</strong>.
+     *
+     * @param recordSchema the format of the data of the records
+     */
+    public void setRecordSchema(final String recordSchema) {
+        this.recordSchema = recordSchema;
+    }
+
+    /**
+     * Sets the kind of operation of the lookup. <strong>Default value: {@value OPERATION}</strong>.
+     *
+     * @param operation the kind of operation of the lookup
+     */
+    public void setOperation(final String operation) {
+        this.operation = operation;
+    }
+
+    /**
+     * Sets the version of the lookup. <strong>Default value: {@value VERSION}</strong>.
+     *
+     * @param version the version of the lookup
+     */
+    public void setVersion(final String version) {
+        this.version = version;
+    }
+
+    @Override
+    public void process(final String baseUrl) {
+
+        final StringBuilder srUrl = new StringBuilder(baseUrl);
+        if (query != null) {
+            srUrl.append("?query=").append(query).append("&operation=").append(operation).append("&recordSchema=")
+                .append(recordSchema).append("&version=").append(version).append("&maximumRecords=" + maximumRecords);
+        }
+        else {
+            throw new IllegalArgumentException("Missing mandatory parameter 'query'");
+        }
+        int recordsRetrieved = 0;
+        int numberOfRecords = Integer.MAX_VALUE;
+        while (!stopRetrieving && recordsRetrieved < totalRecords && startRecord < numberOfRecords) {
+
+            try {
+                final InputStream inputStreamOfURl = retrieveUrl(srUrl);
+                final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                final DocumentBuilder docBuilder = factory.newDocumentBuilder();
```

Comment:
Can the `DocumentBuilder` be reused? Then move it outside the loop or make it `static`.